### PR TITLE
Migrate to request specs in `/api/v1/blocks`

### PR DESCRIPTION
**Pull request by danielmbrasil**

### CODE REVIEW

File: spec/controllers/api/v1/blocks_controller_spec.rb [hunk 1]
```diff
@@ -1,65 +0,0 @@
...
-      get :index, params: { since_id: blocks[0] }
-
-      expect(body_as_json.size).to eq 1
-      expect(body_as_json[0][:id]).to eq blocks[1].target_account_id.to_s
-    end
-
-    it 'sets pagination header for next path' do
-      blocks = Array.new(2) { Fabricate(:block, account: user.account) }
-      get :index, params: { limit: 1, since_id: blocks[0] }
-      expect(response.headers['Link'].find_link(%w(rel next)).href).to eq api_v1_blocks_url(limit: 1, max_id: blocks[1])
```
Comment (mjankowski) [inline spec/controllers/api/v1/blocks_controller_spec.rb:42]: I see `prev` in the new file, did `next` get dropped?

Comment (danielmbrasil) [reply spec/controllers/api/v1/blocks_controller_spec.rb:42]: I'm working on adding better specs for the pagination headers now.

File: spec/controllers/api/v1/blocks_controller_spec.rb [hunk 2]
```diff
@@ -1,65 +0,0 @@
...
-    end
-
-    it 'queries blocks in range according to since_id' do
-      blocks = Array.new(2) { Fabricate(:block, account: user.account) }
-
-      get :index, params: { since_id: blocks[0] }
-
-      expect(body_as_json.size).to eq 1
-      expect(body_as_json[0][:id]).to eq blocks[1].target_account_id.to_s
-    end
```
Comment (ClearlyClaire) [inline spec/controllers/api/v1/blocks_controller_spec.rb:37]: Those do not get tested anymore

Comment (danielmbrasil) [reply spec/controllers/api/v1/blocks_controller_spec.rb]: Good catch. I just included the missing examples.

**Review by ClearlyClaire — CHANGES_REQUESTED** [review]
Could you update it to delete the controller spec this replaces?

**Comment by danielmbrasil** [comment]
> Could you update it to delete the controller spec this replaces?

Done, I'd accidentally included it in the previous rebase 😅.

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**